### PR TITLE
Support nullable vs required

### PR DESCRIPTION
**Pull request by bhelx**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -55,7 +55,7 @@ function toPythonType(property: Property): string {
   }
 
   if (!tp) throw new Error("Cant convert property to Python type: " + property.type)
-  if (!property.nullable) return tp
+  if (!property.nullable && !property.required) return tp
```
Comment (bhelx) [inline src/index.ts:58]: Trying to decide what the logic should be here and if extism.Json is able to distinguish b/w null and optional.

Comment (bhelx) [reply src/index.ts:58]: @zshipko is there an equivalent of `omitempty` for the json encoder? We may want to skip the key if the value is None and the type is !required

Comment (zshipko) [reply src/index.ts:58]: not really, I think the best way to handle nullable is to make the field `Optional[T]` and for non-required we can do the same, but a default value for that field in the dataclass. 

I haven't totally thought it through but that should work.

**Review by zshipko — APPROVED** [review]
There doesn't seem to be a whole lot more we can do without something like omitempty - so this seems good for now.

I have some related changes in https://github.com/dylibso/xtp-python-bindgen/pull/11 that I can update to work better with this change once it's merged.

**Pull request merged**